### PR TITLE
wrapped the db.download(bugzilla.BUGS_DB) line with a try-except block

### LINKED ISSUE #2255: The bug retriever script should not fail if the db is not available yet
E.g. when we bump the version of the DB and we need to regenerate it from scratch.

https://github.com/mozilla/bugbug/blob/c7d2737ee78a7f52f6105738f92884e9c394913a/scripts/bug_retriever.py#L24

### PR DESCRIPTION
**Pull request by rajdeep988**
- fixes #2255
**wrapped the db.download(bugzilla.BUGS_DB) line with a try-except block and catching the db.DBNotAvailable exception**

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
@rajdeep988 since this is still a work in progress, could you please convert it to a draft pull request? Once it is ready for review, you can convert it back.

**Comment by rajdeep988** [comment]
Sure! **Drafted**

**Comment by rajdeep988** [comment]
Hey sir! @suhaibmujahid Could you please review it.

File: scripts/bug_retriever.py [hunk 1]
```diff
@@ -18,13 +18,12 @@ def retrieve_bugs(self, limit: int = None) -> None:
...
-
-        # Get IDs of bugs changed since last run.
         try:
+            db.download(bugzilla.BUGS_DB)
             last_modified = db.last_modified(bugzilla.BUGS_DB)
-        except db.LastModifiedNotAvailable:
-            pass
+        except Exception as e:
+            logger.error(f"Failed to download or access bugs DB: {str(e)}")
+            last_modified = None
```
Comment (suhaibmujahid) [inline scripts/bug_retriever.py:26]: Why do we need this here? The value is already `None`.

File: scripts/bug_retriever.py [hunk 2]
```diff
@@ -18,13 +18,12 @@ def retrieve_bugs(self, limit: int = None) -> None:
...
         last_modified = None
-        db.download(bugzilla.BUGS_DB)
-
-        # Get IDs of bugs changed since last run.
         try:
+            db.download(bugzilla.BUGS_DB)
             last_modified = db.last_modified(bugzilla.BUGS_DB)
-        except db.LastModifiedNotAvailable:
-            pass
+        except Exception as e:
```
Comment (suhaibmujahid) [inline scripts/bug_retriever.py:24]: It is better to avoid general exceptions.

File: scripts/bug_retriever.py [hunk 3]
```diff
@@ -18,13 +18,12 @@ def retrieve_bugs(self, limit: int = None) -> None:
         bugzilla.set_token(get_secret("BUGZILLA_TOKEN"))
 
         last_modified = None
-        db.download(bugzilla.BUGS_DB)
-
-        # Get IDs of bugs changed since last run.
         try:
+            db.download(bugzilla.BUGS_DB)
```
Comment (suhaibmujahid) [inline scripts/bug_retriever.py:22]: Handling two errors in one try-catch is not good practice. Also, I don't think we need to add `db.download()` in a try-catch. The issue mentions failure in `db.last_modified()`, not ``db.download()``.

**Review by suhaibmujahid — COMMENTED** [review]
I think this issue is already fixed in #2665. @marco-c could you please confirm?

**Comment by marco-c** [comment]
> I think this issue is already fixed in #2665. @marco-c could you please confirm?

Yeah, maybe #2665 fixed it. You can test easily by bumping the version of the DB and trying to run the script.

**Comment by suhaibmujahid** [comment]
> > I think this issue is already fixed in #2665. @marco-c could you please confirm?
> 
> Yeah, maybe #2665 fixed it. You can test easily by bumping the version of the DB and trying to run the script.

@marco-c  Thank you! I did that, but I wanted to double check with you.

Sorry, @rajdeep988! I will close this since the issue is already fixed by another issue's fix. Thank you for your efforts.

**Pull request closed without merging**